src/generic/grid.cpp

```diff
@@ -3859,11 +3862,75 @@ void wxGrid::PrepareDCFor(wxDC &dc, wxGridWindow *gridWindow)
     dc.SetDeviceOrigin(dcOrigin.x, dcOrigin.y);
 }
 
+void CheckDoDragScroll(wxGrid *grid, wxGridSubwindow *eventGridWindow,
+    wxGridSubwindow *gridWindow, wxPoint event_pos,
+    bool horizontal, bool vertical)
+{
+    // scroll when at the edges or outside the window
+    // eventGridWindow: the window that received the mouse event
+    // gridWindow: the same or the corresponding non-frozen window
+    int w, h;
+    eventGridWindow->GetSize(&w, &h);
+    // ViewStart is scroll position in scroll units
+    wxPoint sxy = grid->GetViewStart();
+    int sx_new = wxDefaultCoord, sy_new = wxDefaultCoord;
```

Comment:
It looks like the code might be more clear if we used a single `wxPoint ptScroll` or `newOrigin` and then checked if it's different from `ptStart`/`origin` below instead of comparing with `wxDefaultCoord`.

We could then just do `newOrigin.x++` etc instead of writing `sx_new = sxy.x + 1` which is less obvious.